src/Microsoft.Extensions.DependencyModel/DependencyContextWriter.cs

```diff
@@ -203,13 +203,11 @@ private JObject WriteTargetLibrary(Library library)
             {
                 var libraryObject = new JObject();
                 AddDependencies(libraryObject, runtimeLibrary.Dependencies);
-                AddRuntimeAssemblies(libraryObject, runtimeLibrary.Assemblies);
-                AddResourceAssemblies(libraryObject, runtimeLibrary.ResourceAssemblies);
 
-                if (runtimeLibrary.NativeLibraries.Any())
-                {
-                    libraryObject.Add(DependencyContextStrings.NativeLibrariesKey, WriteAssetList(runtimeLibrary.NativeLibraries));
-                }
+                // Add runtime-agnostic assets
+                AddAssets(libraryObject, DependencyContextStrings.RuntimeAssembliesKey, runtimeLibrary.RuntimeAssemblyGroups.FirstOrDefault(r => string.IsNullOrEmpty(r.Runtime)));
```

Comment:
Make a method out of `RuntimeAssemblyGroups.FirstOrDefault(r => string.IsNullOrEmpty(r.Runtime)`